File: app/src/main/java/com/xia/banner/option/img/ImageHolder.java
package com.xia.banner.option.img;

import android.content.Context;
import android.view.View;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.xia.banner.R;
import com.xia.flybanner.holder.FBHolder;

import androidx.appcompat.widget.AppCompatImageView;

/**
 * @author weixia
 * @date 2019/4/16.
 */
public class ImageHolder extends FBHolder {
    private AppCompatImageView mAppCompatImageView;

    ImageHolder(View itemView) {
        super(itemView);
    }

    @Override
    protected void initView(View itemView) {
        mAppCompatImageView = itemView.findViewById(R.id.item_banner_iv);
    }

    @Override
    public void updateUI(Object data) {
        if (mAppCompatImageView == null) {
            return;
        }
        final Context context = mAppCompatImageView.getContext();
        Glide.with(context)
                .load(data)
                .transition(DrawableTransitionOptions.withCrossFade())
                .into(mAppCompatImageView);
    }
}

File: app/src/main/java/com/xia/banner/option/img/ImageHolder.kt
package com.xia.banner.option.img

import android.view.View
import androidx.appcompat.widget.AppCompatImageView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.xia.banner.R
import com.xia.flybanner.holder.FBHolder

/**
 * @author weixia
 * @date 2019/4/16.
 */
class ImageHolder internal constructor(itemView: View) : FBHolder<Any>(itemView) {
    private var imageView: AppCompatImageView? = null

    override fun initView(itemView: View) {
        imageView = itemView.findViewById(R.id.item_banner_iv)
    }

    override fun updateUI(data: Any) {
        imageView?.apply {
            Glide.with(context)
                    .load(data)
                    .transition(DrawableTransitionOptions.withCrossFade())
                    .into(this)
        }
    }
}
